src/styles/mixins/responsive.scss

```diff
@@ -1,5 +1,41 @@
 @import 'variables';
 
+@mixin small { // <= 640
+    @media (max-width: #{breakpoints(small)}) {
+        @content;
+    }
+}
+
+@mixin medium {	// 641 - 768
+    @media (min-width: #{breakpoints(small) + 1}) and (max-width: #{breakpoints(medium)}) {
+        @content;
+    }
+}
+
+@mixin medium-up { // >= 641
+    @media (min-width: #{breakpoints(small) + 1}) {
+        @content;
+    }
+}
+
+@mixin medium-down { // <= 767
+    @media (max-width: #{breakpoints(medium) - 1}) {
+        @content;
+    }
+}
+
+@mixin large { // >= 1024
+    @media (min-width: #{breakpoints(large)}) {
+        @content;
+    }
+}
+
+@mixin large-up { // >= 769
+    @media (min-width: #{breakpoints(medium) + 1}) {
+        @content;
+    }
+}
```

Comment:
There's so many breakpoints here I think they're losing their usefulness.  I don't know about you, but I don't think I have a good enough memory to keep so many in my head.  They also seem to be a bit arbitrary, and there's some gaps (for example, `medium-down` cover <= 767, while `large-up` covers >= 769, which leaves 768 uncovered in that gap, although it does fall into the range of `medium` which is 641-768, which also overlaps with `medium-down`, etc ....).

How did you come up with these breakpoints?